src/components/TokenIcon/TokenIcon.tsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react'
+import {Image, StyleSheet, View} from 'react-native'
+
+import {useIsTokenKnownNft, useNftImageModerated, useTokenInfo} from '../../hooks'
+import {SHOW_NFT_GALLERY} from '../../legacy/config'
+import {COLORS} from '../../theme'
+import {YoroiWallet} from '../../yoroi-wallets'
+import {Icon} from '..'
+import {ModeratedNftIcon} from './ModeratedNftIcon'
+
+export const TokenIcon = ({wallet, tokenId}: {wallet: YoroiWallet; tokenId: string}) => {
+  const tokenInfo = useTokenInfo({wallet, tokenId})
+  const isPrimary = tokenInfo.id === wallet.primaryTokenInfo.id
+  const isTokenNft = useIsTokenKnownNft({wallet, fingerprint: tokenInfo.fingerprint})
+
+  if (isPrimary) return <PrimaryIcon />
+  if (isTokenNft && SHOW_NFT_GALLERY) return <NftIcon wallet={wallet} tokenId={tokenInfo.id} />
+
+  if (typeof tokenInfo.logo === 'string' && tokenInfo.logo.length > 0) {
+    if (isBase64(tokenInfo.logo)) {
+      return <Image source={{uri: `data:image/png;base64,${tokenInfo.logo}`}} style={styles.icon} />
+    } else {
+      return <Image source={{uri: tokenInfo.logo}} style={styles.icon} />
+    }
+  }
+
+  return <Placeholder />
+}
+
+const PrimaryIcon = () => (
+  <View style={[styles.icon, styles.primary]}>
+    <Icon.Cardano color="white" height={35} width={35} />
+  </View>
+)
+
+const NftIcon = ({wallet, tokenId}: {wallet: YoroiWallet; tokenId: string}) => {
+  const nftModeratedImage = useNftImageModerated({wallet, nftId: tokenId})
+
+  if (!nftModeratedImage) return <ModeratedNftIcon status="pending" />
+  return <ModeratedNftIcon image={nftModeratedImage.image} status={nftModeratedImage.status} />
+}
+
+const Placeholder = () => (
+  <View style={[styles.icon, styles.placeholder]}>
+    <Icon.Tokens color={COLORS.TEXT_INPUT} size={35} />
+  </View>
+)
+
+const isBase64 = (string) => {
+  return /^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$/i.test(string)
```

Comment:
include source url

Reply:
Do you mean to receive it as `uri` -> with the prefix?

Reply:
What do you mean @wolverineks ?

Reply:
http://www.where-did-you-get-this.com

Reply:
Seems like regex way to check base64 string can cause issues https://github.com/miguelmota/is-base64/issues/10

I have implemented this https://github.com/validatorjs/validator.js/blob/491d9c0eea23f8401b5739803fb8e55c6860b32b/src/lib/isBase64.js